tauri-updater/src/lib.rs

```diff
@@ -1,19 +1,1034 @@
+      Some("win32".into())
+    } else {
+      Some("win64".into())
+    }
+  } else if cfg!(target_os = "freebsd") {
+    Some("freebsd".into())
+  } else {
+    None
+  }
+}
+
+/// Get the extract_path from the provided executable_path
+pub fn extract_path_from_executable(executable_path: &PathBuf, target: &str) -> PathBuf {
+  // Linux & Windows should need to be extracted in the same directory as the executable
+  // C:\Program Files\MyApp\MyApp.exe
+  // We need C:\Program Files\MyApp
+
+  let mut extract_path = executable_path
+    .parent()
+    .map(PathBuf::from)
+    .expect("Can't determine extract path");
+
+  let extract_path_as_string = extract_path.display().to_string();
+
+  // MacOS example binary is in /Applications/TestApp.app/Contents/MacOS/myApp
+  // We need to get /Applications/TestApp.app
+  // todo(lemarier): Need a better way here
+  // Maybe we could search for <*.app> to get the right path
+  if target == "darwin" && extract_path_as_string.contains("Contents/MacOS") {
+    extract_path = extract_path
+      .parent()
+      .map(PathBuf::from)
+      .expect("Unable to find the extract path")
+      .parent()
+      .map(PathBuf::from)
+      .expect("Unable to find the extract path")
+  };
+
+  // We should use APPIMAGE exposed env variable
+  // This is where our APPIMAGE should sit and should be replaced
+  if target == "linux" && env::var_os("APPIMAGE").is_some() {
+    extract_path = PathBuf::from(env::var_os("APPIMAGE").expect("Unable to extract APPIMAGE path"))
+  }
+
+  extract_path
+}
+
+// Return the archive type to save on disk
+fn detect_archive_in_url(path: &str, target: &str) -> String {
+  path
+    .split('/')
+    .next_back()
+    .unwrap_or(&archive_name_by_os(target))
+    .to_string()
+}
+
+// Fallback archive name by os
+// The main objective is to provide the right extension based on the target
+// if we cant extract the archive type in the url we'll fallback to this value
+fn archive_name_by_os(target: &str) -> String {
+  let archive_name = match target {
+    "darwin" | "linux" => "update.tar.gz",
+    _ => "update.zip",
+  };
+  archive_name.to_string()
+}
+
+// Convert base64 to string and prevent failing
+fn base64_to_string(base64_string: &str) -> crate::Result<String> {
+  let decoded_string = &decode(base64_string.to_owned())?;
+  let result = from_utf8(&decoded_string)?.to_string();
+  Ok(result)
+}
+
+// Validate signature
+// need to be public because its been used
+// by our tests in the bundler
+pub fn verify_signature(
+  archive_path: &PathBuf,
+  release_signature: String,
+  pub_key: &str,
+) -> crate::Result<bool> {
+  // we need to convert the pub key
+  let pub_key_decoded = &base64_to_string(pub_key)?;
+  let public_key = PublicKey::decode(pub_key_decoded);
+  if public_key.is_err() {
+    bail!(
+      crate::Error::Updater,
+      "Something went wrong with pubkey decode"
+    )
+  }
+
+  let public_key_ready = public_key.expect("Something wrong with the public key");
```

Comment:
are these `expect` calls necessary? Would be careful to make sure you don't want the code to panic if it doesn't have to.